include/fi_util.h

```diff
@@ -295,6 +295,32 @@ struct util_event {
 int fi_eq_create(struct fid_fabric *fabric, struct fi_eq_attr *attr,
 		 struct fid_eq **eq_fid, void *context);
 
+/*
+ * Buffer Pool
+ */
+
+struct util_buf_pool {
+	size_t size;
+	size_t num;
+	void *memory_region;
+	struct slist list;
+	fastlock_t lock;
```

Comment:
I would not have the buffer pool provider locking.  It is likely to be used while the caller is holding another lock.